asapdiscovery-data/asapdiscovery/data/postera/manifold_artifacts.py

```diff
@@ -69,25 +75,43 @@ def __init__(
         manifold_id_column : str
             The name of the column containing the manifold id
         """
+        self.target = target
         self.molecule_dataframe = molecule_dataframe.copy()
-        self.molecule_set_id = molecule_set_id
-        self.artifact_type = artifact_type
+        self.molecule_set_id_or_name = molecule_set_id_or_name
+        self.bucket_name = bucket_name
+        self.artifact_columns = artifact_columns
+        self.artifact_types = artifact_types
+
+        if moleculeset_api is None:
+            moleculeset_api = MoleculeSetAPI().from_settings(PosteraSettings())
```

Comment:
sorry ive you've been over this already but why aren't we using a pydantic schema here? instead of need an init method and saving everything to the object here

Reply:
No its a good point, this is a v1 object that we modified slightly, we should probably use a factory similar to our new ones but might push to a new PR.

Reply:
#633